plugins/inputs/prometheus/http_service_discovery_test.go

```diff
@@ -0,0 +1,53 @@
+package prometheus
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/influxdata/telegraf/config"
+)
+
+func TestHttpSD(t *testing.T) {
+	testcasePath := filepath.Join("testcases", "service_discovery")
+	configFilename := filepath.Join(testcasePath, "telegraf.conf")
+	expectedResult := filepath.Join(testcasePath, "http-services.json")
+
+	// read expected result
+	result, err := os.ReadFile(expectedResult)
+	require.NoError(t, err)
+
+	// Create a fake API server
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+		if _, err := w.Write(result); err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			t.Error(err)
+			return
+		}
+	}))
```

Comment:
If w.Write() fails, calling w.WriteHeader() after the write attempt is ineffective because the HTTP headers would have already been sent.

```suggestion
	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
    // Verify it's a GET request
    if r.Method != http.MethodGET {
        w.WriteHeader(http.StatusMethodNotAllowed)
        return
    }
    
    w.Header().Set("Content-Type", "application/json")
    if _, err := w.Write(result); err != nil {
        t.Errorf("Failed to write response: %v", err)
        return
    }
}))
```